src/pytest_docker_compose/__init__.py

```diff
@@ -186,4 +184,27 @@ def scoped_containers_fixture(docker_project: Project, request):
         return scoped_containers_fixture
 
 
+class ContainerDict(dict):
+    """A read-only dictionary that re-fetches containers from a docker-compose
+    project dynamically every time they are accessed."""
+
+    def __init__(self, docker_project: Project) -> None:
+        super().__init__()
+        self.docker_project = docker_project
+        for container in docker_project.containers():
+            super().__setitem__(container.name, container)
+
+    def __getitem__(self, key):
```

Comment:
❔ Could we make the type hinting consistent for this class?  I notice that some methods have partial/complete type hints, while others have none.

Reply:
I'm a little sloppy with my type hints. I will fix it.